src/paths.ts

```diff
@@ -128,28 +142,70 @@ class Paths {
         workspaceRootFolderName: path.basename(wsc.folder.uri.fsPath),
       },
     });
+
     if (raw == 'auto' || raw == 'cmake') {
       // We start by searching $PATH for cmake
       const on_path = await this.which('cmake');
-      if (!on_path) {
+      if (!on_path && (process.platform === 'win32')) {
         if (raw == 'auto' || raw == 'cmake') {
           // We didn't find it on the $PATH. Try some good guesses
-          const candidates = [
+          const default_cmake_paths = [
             `C:\\Program Files\\CMake\\bin\\cmake.exe`,
             `C:\\Program Files (x86)\\CMake\\bin\\cmake.exe`,
           ];
-          for (const cand of candidates) {
-            if (await fs.exists(cand)) {
-              return cand;
+          for (const cmake_path of default_cmake_paths) {
+            if (await fs.exists(cmake_path)) {
+              return cmake_path;
             }
           }
+
+          // Look for bundled CMake executables in Visual Studio install paths
+          const bundled_tools_paths = await this.vsCMakePaths();
+          if (null !== bundled_tools_paths.cmake) {
+            this._ninjaPath = bundled_tools_paths.ninja;
+
+            return bundled_tools_paths.cmake;
+          }
         }
+
         return null;
       }
+
       return on_path;
     }
+
     return raw;
   }
+
+  async vsCMakePaths(): Promise<VSCMakePaths> {
+    const vsCMakePaths = {} as VSCMakePaths;
+
+    const vs_installations = await vsInstallations();
+    if (vs_installations.length > 0) {
+      const bundled_tool_paths = [] as {cmake: string, ninja: string}[];
+
+      for (const install of vs_installations) {
+        const bundled_tool_path = {
+          cmake: install.installationPath + '\\Common7\\IDE\\CommonExtensions\\Microsoft\\CMake\\CMake\\bin\\cmake.exe',
+          ninja: install.installationPath + '\\Common7\\IDE\\CommonExtensions\\Microsoft\\CMake\\Ninja\\ninja.exe'
+        };
+        bundled_tool_paths.push(bundled_tool_path);
+      }
+
+      for (const tool_path of bundled_tool_paths) {
+        if (await fs.exists(tool_path.cmake)) {
+          // Append the bundled Ninja build system path to the VSCode's process' environment variable 'PATH'
+          if (await fs.exists(tool_path.ninja)) {
+            vsCMakePaths.ninja = tool_path.ninja;
+          }
+          // CMake can be still used without Ninja
+          vsCMakePaths.cmake = tool_path.cmake;
```

Comment:
Should this break out of the loop once you find one that works?

Reply:
Yes, that was the intention. Thank you.